### PR TITLE
Feature: Draco PNTS proposal alternative

### PR DESCRIPTION
**Pull request by gkjohnson**
Related to #326 

@Gmadges What do you think of this? I've based this off your previous commit and changed it a bit to consolidate all the decoding logic in the PNTS class.

I haven't tested it, though, so I'll need your help with that.

### CODE REVIEW

**Comment by Gmadges** [comment]
@gkjohnson Looks good to me! A bit cleaner than my PNTSFeatureTable refactor.

I'll give it a test, and let you know how it goes

**Comment by Gmadges** [comment]
Tested, Requires a few tweaks and then should work 👍

File: src/three/PNTSLoader.js [hunk 1]
```diff
@@ -12,68 +23,115 @@ export class PNTSLoader extends PNTSLoaderBase {
...
-					'NORMAL',
-					'NORMAL_OCT16P',
-				].forEach( feature => {
+				// map PNTS keys to draco types
+				const attributeIDs = {};
+				for ( const key in properties ) {
 
-					if ( feature in featureTable.header ) {
+					const mappedKey = DRACO_ATTRIBUTE_MAP[ key ];
+					if ( mappedKey in attributeIds ) {
```
Comment (Gmadges) [inline src/three/PNTSLoader.js:50]: I think this should be 
```
if ( ! ( mappedKey in attributeIDs ) )
```

File: src/three/PNTSLoader.js [hunk 2]
```diff
@@ -12,68 +23,115 @@ export class PNTSLoader extends PNTSLoaderBase {
...
+					attributeTypes: {
+						position: 'Float32Array',
+						color: 'Uint8Array',
+					},
+					useUniqueIDs: true,
+				};
+
+				const buffer = featureTable.getDracoBuffer( byteOffset, byteLength );
+				const geometry = await dracoLoader.decodeGeometry( buffer, taskConfig );
+				geometry.copy( dracoGeometry );
```
Comment (Gmadges) [inline src/three/PNTSLoader.js:70]:
```
const buffer = featureTable.getBuffer( byteOffset, byteLength );
geometry = await dracoLoader.decodeGeometry( buffer, taskConfig );
```

**Review by Gmadges — CHANGES_REQUESTED** [review]
Tested, Requires a few tweaks and then should work 👍

File: src/three/PNTSLoader.js (hunk 2 above)
Comment (gkjohnson) [reply src/three/PNTSLoader.js:70]: Thanks!

File: src/three/PNTSLoader.js (hunk 1 above)
Comment (gkjohnson) [reply src/three/PNTSLoader.js:50]: Oh oops - you're right this isn't correct. But I think it should be the following to make sure both fields are defined:

```js
if ( key in DRACO_ATTRIBUTE_MAP && key in properties ) {

  // ...

} 
```

**Comment by gkjohnson** [comment]
> Tested, Requires a few tweaks and then should work 👍

Thanks! Just made the changes. Can you give it another test to make sure it looks right and then we can get this merged!

File: src/three/PNTSLoader.js [hunk 3]
```diff
@@ -12,68 +23,115 @@ export class PNTSLoader extends PNTSLoaderBase {
...
+				const taskConfig = {
+					attributeIDs,
+					attributeTypes: {
+						position: 'Float32Array',
+						color: 'Uint8Array',
+					},
+					useUniqueIDs: true,
+				};
+
+				const buffer = featureTable.getDracoBuffer( byteOffset, byteLength );
```
Comment (Gmadges) [inline src/three/PNTSLoader.js:68]: `getDracoBuffer` -> `getBuffer`

**Review by Gmadges — CHANGES_REQUESTED** [review]
Sorry, some minor items, and it works 👍

**Comment by gkjohnson** [comment]
> Sorry, some minor items, and it works 👍

Just made another update. If there are more issues do you mind making a PR into this "draco-pnts-proposal" branch?

**Review by Gmadges — APPROVED** [review]
Tested, and all works 👍

**Pull request merged**